cmd/argocd/commands/app.go

```diff
@@ -382,17 +383,26 @@ func NewApplicationGetCommand(clientOpts *argocdclient.ClientOptions) *cobra.Com
   		`),
 
 		Run: func(c *cobra.Command, args []string) {
-			ctx := c.Context()
+			ctx, cancel := context.WithCancel(c.Context())
 			if len(args) == 0 {
 				c.HelpFunc()(c, args)
 				os.Exit(1)
 			}
 			acdClient := headless.NewClientOrDie(clientOpts, c)
 			conn, appIf := acdClient.NewApplicationClientOrDie()
-			defer argoio.Close(conn)
+			defer func() {
+				argoio.Close(conn)
+				cancel()
+			}()
 
 			appName, appNs := argo.ParseFromQualifiedName(args[0], appNamespace)
 
+			if timeout != 0 {
+				time.AfterFunc(time.Duration(timeout)*time.Second, func() {
+					cancel()
+				})
+			}
```

Comment:
```suggestion
                   app, _, err := waitOnApplicationStatus(ctx, acdClient, appName, uint(timeout), watchOpts{operation: true}, nil, output)
                   errors.CheckError(err)
```

You don't need all of this. Instead call the `waitOnApplicationStatus` before making the first GET call.

Reply:
Yeah sounds good. I wanted to ask about `hardRefresh` since waitOnApplicationStatus only do a normal refresh 
please correct me if got it wrong

Reply:
I guess (correct me if i'm wrong) if we proceed with `waitOnApplicationStatus` we would have to flag an `hardRefresh` somehow. maybe a context key-value or adding a parameter causing us to adjust all callers to this function. let me know what do you think and i'll get on it.